src/CRUD.php

```diff
@@ -84,52 +103,129 @@ public function setModel(\atk4\data\Model $m, $defaultFields = null)
-                $form->setModel($m, $this->fieldsCreate ?: $this->fieldsDefault);
-                $form->onSubmit(function ($form) {
-                    $form->model->save();
-
-                    return [
-                        (new jQuery($this))->trigger('reload'),
-                        new jsExpression('$(".atk-dialog-content").trigger("close")'),
-                        $this->notify,
-                    ];
-                });
-            });
+            $this->initCreate();
         }
 
         if ($this->canUpdate) {
-            $this->addAction(['icon' => 'edit'], new jsModal('Edit', $this->pageUpdate, [$this->name => $this->jsRow()->data('id')]));
-
-            $this->pageUpdate->set(function () {
-                $this->model->load($this->app->stickyGet($this->name));
-                $this->formUpdate->setModel($this->model);
-                $this->formUpdate->onSubmit(function ($form) {
-                    $form->model->save();
-
-                    return [
-                        (new jQuery($this))->trigger('reload'),
-                        new jsExpression('$(".atk-dialog-content").trigger("close")'),
-                        $this->notify,
-                    ];
-                });
-            });
+            $this->initUpdate();
         }
 
         if ($this->canDelete) {
-            $this->addAction(['icon' => 'red trash'], function ($j, $id) {
-                $this->model->load($id)->delete();
+            $this->initDelete();
+        }
 
-                return $j->closest('tr')->transition('fade left');
-            }, 'Are you sure?');
+        return $this->model;
+    }
+
+    /**
+     * Initializes interface elements for new record creation
+     */
+    function initCreate()
+    {
+        // setting itemCreate manually is possible.
+        if (!$this->itemCreate) {
+            $this->itemCreate = $this->menu->addItem(
+                $this->itemCreate ?: ['Add new', 'icon' => 'plus'],
+                new jsModal('Add new', $this->pageCreate)
+            );
+            $this->itemCreate->set('Add New '.(isset($this->model->title) ? $this->model->title : preg_replace('|.*\\\|', '', get_class($this->model))));
         }
 
-        return $m;
+        // setting callback for the page
+        $this->pageCreate->set(function () {
+
+            // formCreate may already be an object added inside pageCreate
+            if (!is_object($this->formCreate) || !$this->formCreate->_initialized) {
+                $this->formCreate = $this->pageCreate->add($this->formCreate ?: $this->formDefault);
+            }
+
+           // $form = $page->add($this->formCreate ?: ['Form', 'layout' => 'FormLayout/Columns']);
+            $this->formCreate->setModel($this->model, $this->fieldsCreate ?: $this->fieldsDefault);
+
+            // set save handler with reload trigger
+            $this->formCreate->onSubmit(function ($form) {
+                $this->formCreate->model->save();
```

Comment:
shouldn't use `$form` here instead of `$this->formCreate` ?

Reply:
Technically it's the same, but ill change.